routes/conversation.js

```diff
@@ -255,6 +255,39 @@ exports.leave = function (req, res){
 	});
 };
 
+/**
+ * Action when a user archives a conversation.
+ */
+exports.archive = function (req, res){
+	if (!req.user || !req.user._id){
+		resError(res, "Access denied.", "/error");
+		return;
+	}
+
+	// Search through the user's conversation list until the conversation with the correct
+	// id is found. 
+	User.findById( req.user._id, function (err, user){
+		if (err){
+			reject(res, "Could not find the user.");
```

Comment:
I think the `reject(...)` here and on line 287 should actually be `resError(...)` calls.  Reject is used inside some promise functions, while resError is used for responding to the request with a error.